### PR TITLE
[DOC] fix edit button for examples

### LINKED ISSUE #3977: fix target of edit button for the examples in the doc
currently it directs to (for example)

https://github.com/nilearn/nilearn/edit/main/doc/auto_examples/05_glm_second_level/plot_second_level_association_test.rst

it should be 

https://github.com/nilearn/nilearn/edit/main/examples/05_glm_second_level/plot_second_level_association_test.py

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #3977

Changes proposed in this pull request:

- make sure that the "edit this page" links to the correct file on github for the example
- uniformize name of README files in the examples subfolders

### CODE REVIEW

File: doc/templates/components/edit-this-page.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+{% extends "basic-ng/components/edit-this-page.html" %}
+{% from "basic-ng/components/edit-this-page.html" import determine_page_edit_link with context %}
+
+{%- macro furo_edit_button(url) -%}
+<!-- {%- if "auto_examples" in url -%}
+{{ url }}
+{%- endif -%} -->
+<div class="edit-this-page">
+  <a class="muted-link" href="{{ url|replace("doc/auto_examples", "examples")|replace(".rst", ".py") }}" title="{{ _("Edit this page") }}">
```
Comment (Remi-Gau) [inline doc/templates/components/edit-this-page.html:11]: This is pretty much the only line that was changed compared to the original furo component.

Comment (Remi-Gau) [reply doc/templates/components/edit-this-page.html:11]: Pronably worth adding a comment in this file explaining why it's there.

Comment (Remi-Gau) [reply doc/templates/components/edit-this-page.html]: > Pronably worth adding a comment in this file explaining why it's there.

Done

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by ymzayek** [comment]
@Remi-Gau was the intention here to prefer to link to the examples for editing over rst files in e.g. `doc/`? Because now those are broken because extention .rst is replaced with .py (take the glossary for example). Is there a way to fix the edit button conditionally?

**Comment by Remi-Gau** [comment]
> @Remi-Gau was the intention here to prefer to link to the examples for editing over rst files in e.g. `doc/`? Because now those are broken because extention .rst is replaced with .py (take the glossary for example). Is there a way to fix the edit button conditionally?

Good catch, need to improve the jinja template. You are correct that I may have to use some conditionals.

File: doc/templates/components/edit-this-page.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  Required to properly link the edit button in the examples from the doc to source document in the github repo.
+#}
+
+{% extends "basic-ng/components/edit-this-page.html" %}
+{% from "basic-ng/components/edit-this-page.html" import determine_page_edit_link with context %}
+
+{%- macro furo_edit_button(url) -%}
+{%- if 'doc/auto_examples' in url -%}
+{% set url=url|replace("doc/auto_examples", "examples")|replace(".rst", ".py")|replace("index.py", "README.rst") %}
+{%- endif -%}
```
Comment (Remi-Gau) [inline doc/templates/components/edit-this-page.html:12]: @ymzayek 
this seemed to do the trick locally

Comment (ymzayek) [reply doc/templates/components/edit-this-page.html:12]: Just checked it and it works! Thanks a bunch

Comment (Remi-Gau) [reply doc/templates/components/edit-this-page.html:12]: thanks for spotting the issue and checking

**Pull request merged**